app/controllers/orderitems_controller.rb

```diff
@@ -0,0 +1,48 @@
+class OrderitemsController < ApplicationController
+
+  #this method hooks up to button in cart to delete an orderitem, OR you might end up redirected here if you update at orderitem quantity to zero.
+  #if you are deleting the last item in the cart, the whole order gets deleted
+  def destroy
+    @orderitem = Orderitem.find(params[:id])
+    @order = @orderitem.order
+    total_items = @order.orderitems
+    if total_items.count == 1
+      @order = Order.find(session[:order_id])
+      @order.destroy!
+      session[:order_id] = nil
+      redirect_to root_path
+    else
+      @orderitem.destroy
+      redirect_to cart_path
```

Comment:
Again, do these need to be instance variables? If you're using these variables outside of the method (like in your view) they should be instance variables. But if they're being used only in the method to store data, they don't need to be instance variables.